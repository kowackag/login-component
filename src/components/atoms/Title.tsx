import { ReactNode } from "react";

export const Title = ({ children }: { children: ReactNode }) => {
	return <h1 className="mb-5 text-lg font-bold uppercase">{children}</h1>;
};
