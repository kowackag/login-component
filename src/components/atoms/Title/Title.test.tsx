import { expect, it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import { Title } from "./Title";

describe("Title", () => {
	it("Title component displays appropriate text and has correct role", () => {
		render(<Title text="Login page" level="h2" />);

		const title = screen.getByRole("heading", { name: "Login page" });
		expect(title).toBeVisible();
	});

	it("Title component displays correct heading level", () => {
		render(<Title text="Login page" level="h4" />);
		const title = screen.getByRole("heading", { level: 4 });
		expect(title).toBeVisible();
	});
});
