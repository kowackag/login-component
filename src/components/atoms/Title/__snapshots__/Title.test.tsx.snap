// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Title > renders correctly 1`] = `
<div>
  <h2
    class="mb-4 text-lg font-bold uppercase"
    data-testid="title"
  >
    Login page
  </h2>
</div>
`;
